packages/core/src/Stores/pushwoosh-store.js

```diff
@@ -19,6 +19,18 @@ export default class PushwooshStore extends BaseStore {
      */
     @action.bound
     init = () => {
+        let reaction_id;
+        if (!this.root_store.common.is_network_online) {
+            reaction_id = when(
+                () => this.root_store.common.is_network_online,
+                () => {
+                    this.init();
+                }
+            );
+            return;
+        }
+        // canceling the reaction once is_network_online becomes true for the first time:
+        reaction_id?.cancel();
         if (!this.is_applicable && this.has_initialized) return;
```

Comment:
Can we move this line to the top? 
So that, if `is_applicable` is false or `has_initialized` is true then we don't need to setup a `when` reaction.